lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -586,6 +599,18 @@ def get_thread_list(
     course = _get_course(course_key, request.user)
     context = get_context(course, request)
 
+    author_id = None
```

Comment:
nit: this seems unnecessary

Reply:
The `author_id` is only set in a conditional inside `if author` so the variable needs to be set to `None` for other cases.